Code/GraphMol/ChemTransforms/ChemTransforms.cpp

```diff
@@ -358,11 +375,45 @@ ROMol *replaceCore(const ROMol &mol, const ROMol &core,
       throw ValueErrorException(
           "Supplied MatchVect indices out of bounds of the target molecule");
     }
-
+    bool useMatch = false;
     if (replaceDummies || core.getAtomWithIdx(mvit.first)->getAtomicNum() > 0) {
       matchingIndices[mvit.second] = mvit.first;
+      useMatch = true;
     }
     allIndices[mvit.second] = mvit.first;
+    SideChainMapping mapping(mvit.second, mvit.first, useMatch);
+    matches.emplace_back(mvit.second, mapping);
+    if (molAtomsMapped[mvit.second]) {
+      multipleMappedMolAtoms.insert(mvit.second);
+    }
+    molAtomsMapped.set(mvit.second);
+  }
+
+  boost::dynamic_bitset<> multipleOwnedBonds(mol.getNumBonds());
+  if (multipleMappedMolAtoms.size() > 0) {
+    for (const auto &match : matches) {
+      const auto &mappingInfo = match.second;
+      if (multipleMappedMolAtoms.find(mappingInfo.molIndex) !=
+          multipleMappedMolAtoms.end()) {
+        auto coreAtom = core.getAtomWithIdx(mappingInfo.coreIndex);
+        CHECK_INVARIANT(
+            coreAtom->getDegree() == 1,
+            "Multiple core atoms match a mol atom, but one of the core "
+            "atoms has degree > 1 ");
+        auto coreNeighborIdx = *core.getAtomNeighbors(coreAtom).first;
```

Comment:
This relies on the boost::graph returning atom indices as vertex descriptors.
It currently works (and probably shows up elsewhere in the code) but would break if we were to make changes to the underlying graph representation.
It would be safer to do:
```
auto coreNeighborIdx = core[*core.getAtomNeighbors(coreAtom).first]->getIdx();
```

Reply:
Changed